Islandora/islandora_internet_archive_bookreader
# Islandora 2436

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2436

# What does this Pull Request do?

Makes the IA Bookviewer view mode default to that determined by islandora internet archive bookviewer module.

# What's new?

The "mode" setting is passed into Drupal.behaviors.islandoraInternetArchiveBookReader by template_preprocess_islandora_internet_archive_bookreader(), it is not utilized anywhere to actually set the bookreader view mode. This pull request passes this setting into bookReader.switchMode() to set the viewing mode for the book reader.  

# How should this be tested?

* In an unmodified islandora instance with internet archive bookreader installed and book objects available to view, you should see that no matter what setting is used for "Default page view" at /admin/islandora/islandora_viewers/internet_archive_bookreader, when you go to /islandora/object/[PID], the view will default to 2-up display.
* After applying this change to islandora_internet_archive_bookreader (and ensuring that drupal's javascript cache is cleared), the book object display will switch to the correct mode - either that defined in the islandora_internet_archive_bookreader viewer settings, or in the viewHint of the paged content object being viewed.


# Additional Notes:

* Does this change the interface, add a new feature, or otherwise change behaviours that would require updating documentation? 
_No_
* Does this change add any new dependencies? 
_No_
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? 
_No_
* Could this change impact execution of existing code?
_No_
# Interested parties
@Islandora/7-x-1-x-committers @DiegoPino
